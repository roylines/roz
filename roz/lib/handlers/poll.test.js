const {handle} = require('./poll');
const {autoscaling} = require('../aws');
const {send} = require('../telegram');

jest.mock('../aws');
jest.mock('../telegram');

describe('poll handler', () => {
  beforeEach(() => jest.resetAllMocks());
  it('should do nothing if no instances returned', async () => {
    autoscaling.get.mockResolvedValue({instance: null});
    await handle();
    expect(autoscaling.get.mock.calls).toMatchSnapshot();
    expect(send).not.toHaveBeenCalled();
    expect(autoscaling.set).not.toHaveBeenCalled();
  });
  it('should do nothing if instances is not running', async () => {
    autoscaling.get.mockResolvedValue({instance: {status: 'terminated'}});
    await handle();
    expect(autoscaling.get.mock.calls).toMatchSnapshot();
    expect(send).not.toHaveBeenCalled();
    expect(autoscaling.set).not.toHaveBeenCalled();
  });
  it('should do nothing if instances has been running for a short time', async () => {
    autoscaling.get.mockResolvedValue({
      instance: {
        status: 'running',
        launched: Date.now(),
      },
    });
    await handle();
    expect(autoscaling.get.mock.calls).toMatchSnapshot();
    expect(send).not.toHaveBeenCalled();
    expect(autoscaling.set).not.toHaveBeenCalled();
  });
  it('should set capacity to 0 if instances has been running for a long time', async () => {
    autoscaling.get.mockResolvedValue({
      instance: {
        status: 'running',
        launched: new Date(2010, 1, 1).getTime(),
      },
    });
    await handle();
    expect(autoscaling.get.mock.calls).toMatchSnapshot();
    expect(send).toMatchSnapshot();
    expect(autoscaling.set).toMatchSnapshot();
  });
});
