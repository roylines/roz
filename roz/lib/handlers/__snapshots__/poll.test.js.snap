// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poll handler should do nothing if instances has been running for a short time 1`] = `
Array [
  Array [],
]
`;

exports[`poll handler should do nothing if instances is not running 1`] = `
Array [
  Array [],
]
`;

exports[`poll handler should do nothing if no instances returned 1`] = `
Array [
  Array [],
]
`;

exports[`poll handler should set capacity to 0 if instances has been running for a long time 1`] = `
Array [
  Array [],
]
`;

exports[`poll handler should set capacity to 0 if instances has been running for a long time 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Array [
          "Bastion has been open for too long.",
          "I'm closing it",
        ],
      ],
    ],
    Array [
      Array [
        Array [
          "Close requested",
          "Watching. Always watching ðŸ‘€",
        ],
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`poll handler should set capacity to 0 if instances has been running for a long time 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "capacity": 0,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
